### PR TITLE
Releases/2.x

### PR DESCRIPTION
**Pull request by Ze-Ra-Xim**
Some changes/fixes we needed to allow proper version calculation. Would be great to see a new 2.x version published with these changes included.

I've also seen usage of replace("/", "\\/") in the branch filter. Just wanted to remark that this replace doesn't work in case the prefix(es) is define as escaped character, e.g.:

`        branchPrefixSeperator = '\/'`

So maybe it would be better to remove those replace() calls and just update documentation for prefix definitions?

### CODE REVIEW

**Comment by m-raab** [comment]
Currently it is not possible to merge this, because tests failed. Unfortunately some of the tests are connected to private test repositories and write access to this repos is not allowed. But you can fork this repos too. It seems that the plugin with your changes fits now your requirements, but this alone will not help.
I will check your changes step by step, but this will take longer.

**Comment by Ze-Ra-Xim** [comment]
> Currently it is not possible to merge this, because tests failed. Unfortunately some of the tests are connected to private test repositories and write access to this repos is not allowed. But you can fork this repos too. It seems that the plugin with your changes fits now your requirements, but this alone will not help.
> I will check your changes step by step, but this will take longer.

That's why I tried to have every single change in a seperate commit, so it is easier to verify/test the changes. So maybe you can cherry-pick change by change to see which one actually breaks the tests?

**Comment by m-raab** [comment]
But that is not how it works. A pull request should be without errors. I will check your commits, but it will take longer before we create a new release.

File: src/main/groovy/com/intershop/gradle/scm/services/git/GitLocalService.groovy [hunk 1]
```diff
@@ -86,6 +86,7 @@ class GitLocalService extends ScmLocalService{
                 setFeatureBranchName((mbb[0] as List)[(mbb[0] as List).size() - 1].toString())
             } else if(msb.matches() && msb.count == 1) {
                 branchType = BranchType.branch
+                withVersion = true
```
[the remark this answers is not in the document]
Comment (Ze-Ra-Xim) [reply src/main/groovy/com/intershop/gradle/scm/services/git/GitLocalService.groovy:91]: I see. I assume setFeatureBranchName(..) is not called in this case for stabilization branch, as that does not contain a name and the call of the method might mess up the branch name. So maybe replacing that line with following suggestion could be fine?

```suggestion
        def versionCheck = branchName =~ /^\d+(\.\d+)?(\.\d+)?(\.\d+)?(-.*)?/
        if (versionCheck.matches()) {
            withVersion = true
        }
```

**Comment by Ze-Ra-Xim** [comment]
@m-raab , does this PR have a chance to get merged into 2.7 release? Is there anything I can do to get it merged?

**Comment by m-raab** [comment]
I think, currently not. The build is red and your change will destroy the api of the plugin. I checked the functionality with a gitflow process and this will not work. I'm writing currently a new plugin for gitflow processes. Maybe you can use the Netflix nebula plugin or an other plugin.

**Comment by Ze-Ra-Xim** [comment]
You have done some commits on the 2.7 branche since I created the PR. But I can (try) to resolve the merge conflicts if that helps. Will you release a new 2.7.x version with the changes you made?

Which APIs does my change brake? Maybe I could rework those then?

**Comment by m-raab** [comment]
> You have done some commits on the 2.7 branche since I created the PR. But I can (try) to resolve the merge conflicts if that helps. Will you release a new 2.7.x version with the changes you made?
> 
> Which APIs does my change brake? Maybe I could rework those then?
I tried to solve the issues from your first commit, but it is not possible to solve the issue. The plugin self will not work correctly with the Gitflow approach. The necessary changes will destroy the currently functionality and other user will have an issue.

**Comment by Ze-Ra-Xim** [comment]
Hi Matthias (@m-raab) ,

I've updated the PR to fix/resolve the conflicts from your changes that you made for 2.7.3 release. Would be great if could review again and if possible merge the PR for a new 2.7.x release.

If any change is not acceptable, then please let me know which and why, and I will try to fix and update PR.

Thanks,
Zeljko

**Comment by m-raab** [comment]
Please see https://travis-ci.org/github/IntershopCommunicationsAG/scmversion-gradle-plugin/builds/743180519 is still red and broken. That means your changes will destroy the "api" and the functionality for other users.

**Comment by Ze-Ra-Xim** [comment]
I just see that some unit test(s) fail. Is there any test report publicily accessible so I can check which unit tests fail?
Or can you help with to get the unit tests executable locall on my dev env? Cause those seem to require some credentials or similar set up to run correctly...

**Pull request closed without merging**